src/parallel.rs

```diff
@@ -2,25 +2,24 @@ use std::sync::{mpsc::sync_channel, Arc, Condvar, Mutex};
 
 // allows producing data in parallel while still consuming it in the main thread
 // in order
-pub fn parallel_process<Iter, Item, Context, Consumer, ContextCreator, Data, Error, Producer>(
+pub fn parallel_process<Iter, Item, Context, Consumer, Data, Error, Producer>(
     iter: Iter,
-    create_thread_context: ContextCreator,
+    context: Context,
     produce: Producer,
     mut consume: Consumer,
 ) -> Result<(), Error>
 where
     Iter: ExactSizeIterator<Item = Item> + Send + 'static,
-    Context: Send + 'static,
-    ContextCreator: Fn() -> Result<Context, Error>,
+    Context: Clone + Send + 'static,
     Data: Send + 'static,
-    Producer: Fn(&mut Context, Item) -> Data + Clone + Send + 'static,
```

Comment:
Context only makes sense as mut, otherwise the user could just capture it in the Producer. Basically what a user cannot do normally is create  mutable resources per thread. Do we still have this use-case? If not: Just remove Context?

Reply:
Solve in the new version of producer.